pgrx/src/toast.rs

```diff
@@ -0,0 +1,44 @@
+use core::ops::{Deref, DerefMut};
+
+pub(crate) enum Toast<T>
+where
+    T: Toasty,
+{
+    Stale(T),
+    Fresh(T),
+}
+
+pub(crate) trait Toasty: Sized {
+    fn detoast(self) -> Toast<Self>;
+    /// Why does it always land butter-side down?
+    fn drop_toast(&mut self);
```

Comment:
we were just discussing this on discord... Seems `drop_toast()` should be `unsafe`